src/nvim/syntax.c

```diff
@@ -7186,20 +7215,27 @@ int syn_namen2id(char_u *linep, int len)
 int syn_check_group(char_u *pp, int len)
 {
   char_u  *name = vim_strnsave(pp, len);
+
+  // HACK we need to make sure that the first highlight is the "Normal" hl
+  if (highlight_ga.ga_len == 0) {
+    int res = syn_add_group((char_u *)"Normal");
```

Comment:
`syn_add_group()` requires its argument to be an allocated string.  You're passing a static string instead.  Try this:

```c
int res = syn_add_group((char_u *)xstrdup("Normal"));
```